libs/vm-modules/src/ml/model/model.cpp

```diff
@@ -35,9 +34,60 @@ namespace vm_modules {
 namespace ml {
 namespace model {
 
-using SizeType    = fetch::math::SizeType;
+using fetch::math::SizeType;
+using fetch::ml::ops::LossType;
+using fetch::ml::OptimiserType;
+using fetch::ml::details::ActivationType;
 using VMPtrString = Ptr<String>;
 
+std::map<std::string, SupportedLayerType> const VMModel::layer_types_{
+    {"dense", SupportedLayerType::DENSE},
+    {"conv1d", SupportedLayerType::CONV1D},
+    {"conv2d", SupportedLayerType::CONV2D},
+};
+
+std::map<std::string, ActivationType> const VMModel::activations_{
+    {"nothing", ActivationType::NOTHING},
+    {"leaky_relu", ActivationType::LEAKY_RELU},
+    {"log_sigmoid", ActivationType::LOG_SIGMOID},
+    {"log_softmax", ActivationType::LOG_SOFTMAX},
+    {"relu", ActivationType::RELU},
+    {"sigmoid", ActivationType::SIGMOID},
+    {"softmax", ActivationType::SOFTMAX},
+    {"gelu", ActivationType::GELU},
+};
+
+std::map<std::string, LossType> const VMModel::losses_{
+    {"mse", LossType::MEAN_SQUARE_ERROR},
+    {"cel", LossType::CROSS_ENTROPY},
+    {"scel", LossType::SOFTMAX_CROSS_ENTROPY},
+};
+
+std::map<std::string, OptimiserType> const VMModel::optimisers_{
+    {"adagrad", OptimiserType::ADAGRAD},   {"adam", OptimiserType::ADAM},
+    {"momentum", OptimiserType::MOMENTUM}, {"rmsprop", OptimiserType::RMSPROP},
+    {"sgd", OptimiserType::SGD},
+};
+
+std::map<std::string, uint8_t> const VMModel::model_categories_{
+    {"none", static_cast<uint8_t>(ModelCategory::NONE)},
+    {"sequential", static_cast<uint8_t>(ModelCategory::SEQUENTIAL)},
+    {"regressor", static_cast<uint8_t>(ModelCategory::REGRESSOR)},
+    {"classifier", static_cast<uint8_t>(ModelCategory::CLASSIFIER)},
+};
+
+template <typename T>
+inline T VMModel::ParseName(std::string const &name, std::map<std::string, T> const &dict,
+                            std::string const &errmsg) const
+{
+  auto found = dict.find(name);
```

Comment:
"auto" is a perfect way to avoid explicit usage of huge typenames, that are necessary to work with STL containers. In this particular case, it would be a 
typename std::map<std::string, T>::const_iterator found = ...
which from my point of view is not so much more clear than a plain "auto" :)

Reply:
UPD: our static analyzer also thinks "auto" is suitable here.

Reply:
refactored without a need for "auto" usage :)